consensus/service/src/api/peer_api_service.rs

```diff
@@ -244,98 +263,61 @@ impl ConsensusPeerApi for PeerApiService {
-                    from_responder_id
-                );
-                return;
-            }
-
-            let unverified_consensus_msg: mc_peers::ConsensusMsg =
-                match deserialize(request.get_payload()) {
-                    Ok(val) => val,
-                    Err(err) => {
-                        send_result(
-                            ctx,
-                            sink,
-                            Err(rpc_invalid_arg_error(
-                                "send_consensus_msg",
-                                format!("Message deserialize error: {}", err),
-                                &logger,
-                            )),
-                            &logger,
-                        );
-                        return;
-                    }
-                };
-
-            // Validate message signature
-            let consensus_msg: mc_peers::VerifiedConsensusMsg = match unverified_consensus_msg
-                .clone()
-                .try_into()
-            {
-                Ok(val) => val,
-                Err(err) => {
-                    log::error!(
-                        logger,
-                        "Signature verification failed for msg {:?} from node_id {:?}: {:?}, disregarding.",
-                        unverified_consensus_msg,
-                        from_responder_id,
-                        err
-                    );
-                    send_result(
-                        ctx,
-                        sink,
-                        Err(rpc_invalid_arg_error(
-                            "send_consensus_msg",
-                            format!("Signature verification failed: {}", err),
-                            &logger,
-                        )),
+            let consensus_msg: mc_peers::ConsensusMsg = match deserialize(request.get_payload()) {
+                Ok(consensus_msg) => consensus_msg,
+                Err(_) => {
+                    let result = Err(rpc_invalid_arg_error(
+                        "send_consensus_msg",
+                        "consensus_msg",
                         &logger,
-                    );
+                    ));
+                    send_result(ctx, sink, result, &logger);
                     return;
                 }
             };
 
-            log::trace!(
-                logger,
-                "received consensus message from {}: {:?}",
-                from_responder_id,
-                consensus_msg,
-            );
+            let result: Result<ConsensusMsgResponse, RpcStatus> = match self
+                .handle_consensus_msg(consensus_msg, from_responder_id)
+            {
+                Ok(()) => {
+                    let mut response = ConsensusMsgResponse::new();
+                    response.set_result(ConsensusMsgResult::Ok);
+                    Ok(response)
+                }
+                Err(peer_service_error) => match peer_service_error {
```

Comment:
No need to nest match statements, you can match `Err(PeerServiceError::UnknownPeer(_))`, `Err(PeerServiceError::ConsensusMsgInvalidSignature)`, `Err(_)`.